### PR TITLE
Prompt require editor restart to user when gizmo color changed

### LINKED ISSUE #82813: Missing prompt to restart editor when 3D Gizmo color editor settings changed
### Godot version

4.1.2

### System information

Windows 10

### Issue description

Changes to `Editor Settings -> Editors -> 3D Gizmos`, which require an editor restart to take effect, do not prompt "The editor must be restarted for changes to take effect.". Contrast this with changes to `Editor Settings -> Editors -> 3D -> Selection Box Color`, which does prompt the user to restart the editor (and helpfully provides a button to do so).

Screenshot from before restarting editor to apply changes. Note the unchanged selection color of the CSG box, and the lack of a prompt to restart the editor.
Godot_v4 1 2-stable_mono_win64_tuwiwlMJnb

### Steps to reproduce

Add a 3D scene. Add a CSGBox3D. Save the scene.
Select the box, observe the default blue tint. Deselect the box.
Change `Editor Settings -> Editors -> 3D Gizmos -> CSG` to `#ff0000`. Observe the lack of a prompt to restart the editor.
Select the box, observe the _unchanged_ default blue tint. Deselect the box.
Restart the editor.
Select the box, observe the correct, changed red tint.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fixes #82813

### CODE REVIEW

File: editor/editor_settings.cpp [hunk 1]
```diff
@@ -1122,8 +1122,8 @@ Variant _EDITOR_DEF(const String &p_setting, const Variant &p_default, bool p_re
 		ret = EDITOR_GET(p_setting);
 	} else {
 		EditorSettings::get_singleton()->set_manually(p_setting, p_default);
-		EditorSettings::get_singleton()->set_restart_if_changed(p_setting, p_restart_if_changed);
 	}
+	EditorSettings::get_singleton()->set_restart_if_changed(p_setting, p_restart_if_changed);
```
Comment (fire) [inline editor/editor_settings.cpp:1126]: Is this required?

**Comment by Calinou** [comment]
`EDITOR_DEF` should be changed for all 3D editor gizmos, not just CSG. (I believe these all require a restart to make the changes visible.)

File: editor/editor_settings.cpp (hunk 1 above)
Comment (jsjtxietian) [reply editor/editor_settings.cpp:1126]: I believe so, as there is a editor setting cache mechanism, we have to set the restart_if_changed if the value is directly read from `EDITOR_GET`.

**Review by akien-mga — APPROVED** [review]

**Comment by YuriSizov** [comment]
Why does this need an editor restart in the first place? This should be something we can update on the fly, surely?

**Comment by akien-mga** [comment]
It could be updated on the fly yes, that's just more work :P

**Comment by YuriSizov** [comment]
Thanks! It doesn't look like we're going to address the restart issue soon, so this is a good PR to have in the meantime.

**Comment by jsjtxietian** [comment]
> It could be updated on the fly yes

Yes it would require recreate the material after the color is changed. If there is a convient way of get notifed after the color is changed in project setting, it'd doable and should be pretty straight forward.

**Comment by YuriSizov** [comment]
Cherry-picked for 4.2.2.

**Pull request merged**